File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
#include <iostream>

using namespace std;

int solution(vector<int> elements) {
    set<int> Set;

    int n = elements.size();

    for (int i = 0 ; i < n ; ++i) {
        int sum = 0;
        for (int j = i ; j < i + n ; ++j) {
            sum += elements[j % n];
            // cout << elements[j % n] << endl;
            Set.insert(sum);
        }
        // cout << "----------" << endl;
    }

    return Set.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(x):
    l = len(x) # 원소의 개수
    new_list = x * 2
    result = []
    for i in range(1,l+1):
        for j in range(l):
            result.append(sum(new_list[j:j+i]))
    return len(set(result))
    